common-custom-user-data-maven-extension/src/main/java/com/gradle/ApiAccessor.java

```diff
@@ -10,31 +9,28 @@
 
 import static java.util.Comparator.comparing;
 
-final class BuildScanApiAccessor {
+final class ApiAccessor {
 
-    private static final String PACKAGE = "com.gradle.maven.extension.api.scan";
-    private static final String BUILD_SCAN_API_CONTAINER_OBJECT = PACKAGE + ".BuildScanApi";
-
-    static BuildScanApi lookup(PlexusContainer container, Class<?> extensionClass) throws MavenExecutionException {
-        ensureBuildScanApiIsAccessible(extensionClass);
-        return lookupBuildScanApi(container);
+    static <T> T lookup(Class<T> componentClass, String componentPackage, String componentRole, PlexusContainer container, Class<?> extensionClass) throws MavenExecutionException {
```

Comment:
❌ `componentPackage` and `componentRole` should be derived from `componentClass`. `componentRole` is just the fully-qualified class name, isn't it?

Reply:
Done